packages/hydra-cli/src/templates/textsearch/service.ts.mst

```diff
@@ -21,28 +26,127 @@ export class {{query.typePrefix}}FTSService {
+            let c = ``;
+
+            if (w.condition.includes(`IN (:...`)) {
+              // IN condition parameters
+              const params: any[] = qb.expressionMap.parameters[`param${index}`];
+
+              // Do nothing when IN condition has an empty list of values
+              if (params.length !== 0) {
+                const paramsAsString = params
+                  .map((_: any) => {
+                    parameterCounter += 1;
+                    return `$${parameterCounter}`;
+                  })
+                  .join(`, `);
+                c = w.condition.replace(`(:...param${index})`, `(${paramsAsString})`);
+              }
+            } else if (w.condition.includes(`->>`)) {
+              parameterCounter += 1;
+              const m = w.condition.match(/->>.*\s=\s:.*/g);
+              if (m === null)
+                throw Error(`Failed to construct where condition for json field: ${w.condition}`);
+              c = w.condition.replace(/=\s:.*/g, `= $${parameterCounter}`);
+            } else {
+              parameterCounter += 1;
+              c = w.condition.replace(`:param${index}`, `$${parameterCounter}`);
+            }
+            WHERE = WHERE.concat(c, ` `, w.type.toUpperCase(), ` `);
+          });
+
+          // Remove unnecessary AND at the end.
+          WHERE = WHERE.slice(0, -4);
+
+          // Add new query to queryString
+          queries = queries.concat(generateSqlQuery(repositories[index].metadata.tableName, WHERE));
+        }
+      });
+
+      queries = `
+  WITH selected_ids AS (`.concat(
+        queries,
+        `
+  )`
+      );
+      return [queries, parameters, parameterCounter];
+    }
+
+    async search(
+      text: string,
+      limit = 5,
+      skip = 0,
+      {{#entities}}where{{type}}?: {{type}}WhereInput,{{/entities}}
+    ): Promise<{{query.typePrefix}}FTSOutput[]> {
+        const wheres = [{{#entities}}where{{type}}, {{/entities}}]
+        let [queries, parameters, parameterCounter]: [string, any[], number] = [``, [], 0];
+
+        if (wheres.some(f => f !== undefined)) {
+          [queries, parameters, parameterCounter] = await this.processWheres(wheres);
+        }
+        parameters.push(...[text, limit, skip]);
+
+        return getConnection().transaction<{{query.typePrefix}}FTSOutput[]>(
+          'REPEATABLE READ',
+          async (em: EntityManager) => {
+            const query = `
+              ${queries}
+              SELECT origin_table, id, 
+                  ts_rank(tsv, phraseto_tsquery('english', $${parameterCounter + 1})) as rank,
+                  ts_headline(document, phraseto_tsquery('english', $${parameterCounter +
+                    1})) as highlight
+              FROM comment_search_view
+              WHERE phraseto_tsquery('english', $${parameterCounter + 1}) @@ tsv
+              ${
+                queries !== ``
+                  ? `AND origin_table || '_' || id IN (SELECT unique_id FROM selected_ids)`
```

Comment:
This query is going to be slow. I suggest adding `origin_table || '_' || id` to `comment_search_view` as a separate field (say, `unique_id`) and add an index to it. Then it will be fast